### PR TITLE
feat(rn-camera): deviceOrientation, videoOrientation, pictureOrientation

### PR DESCRIPTION
**Pull request by n1ru4l**
- [x] ios
- [x] android
- [x] documentation

Fixes https://github.com/react-native-community/react-native-camera/issues/1983

Features:

- implements `orientation` option on `recordAsync` and `takePictureAsync`
- adds field `deviceOrientation` and `videoOrientation` to `recordAsync` result
- adds field `deviceOrientation` and `pictureOrientation` to `takePictureAsync` result

### CODE REVIEW

File: ios/RN/RNCamera.m [hunk 1]
```diff
@@ -522,7 +531,9 @@ - (void)record:(NSDictionary *)options resolve:(RCTPromiseResolveBlock)resolve r
         }
     }
 
-    [self updateSessionAudioIsMuted:[options[@"mute"] boolValue]];
+    if ([options valueForKey:@"mute"] != nil) {
```
Comment (n1ru4l) [inline ios/RN/RNCamera.m]: I also fixed this regression so it prevents the flickering if no mute property is present in `options["mute"]`. Nevertheless we should implement `defaultMuted` asap.

**Comment by n1ru4l** [comment]
@sibelius This is only for capturing the initial device orientation when starting the recording/taking the picture. This will not include orientation changes over time. Nevertheless that information could be quite useful for cases where you want to train a machine learning model, so we cloud implement it in a future pull request (once somebody needs that information).

**Comment by n1ru4l** [comment]
I will now try to implement the android counterpart

**Comment by sibelius** [comment]
If orientation changes over time, does the width and height of video changes as well? This could be enough to check device orientation

**Comment by n1ru4l** [comment]
@sibelius I also checked it on android and ran into this bug: https://github.com/react-native-community/react-native-camera/issues/1988

**Comment by sibelius** [comment]
have you tried to use PermissionsAndroid ?

try with latest version of react native

permissions android had some bug with promises before

**Comment by n1ru4l** [comment]
@sibelius I don't understand your suggestion. When I record without rotation everything works fine. This should not be related to permissions  or am I missing out on something?

**Comment by sibelius** [comment]
try to add some breakpoints on android studio to check where the promise is broke

**Comment by n1ru4l** [comment]
I now implemented the same features on android + implemented the `orientation` option which was documented as implemented on both platforms but actually missing on android.

**Pull request merged**